packages/block-editor/src/store/actions.js

```diff
@@ -1111,6 +1111,24 @@ export const mergeBlocks =
 			return;
 		}
 
+		// If previous block is empty, remove previous block and update selection.
+		const blockAHasEmptyContent = blockA?.attributes.content === '';
+		const blockBHastEmptyContent = blockB?.attributes.content === '';
```

Comment:
Maybe we should use the `isUnmodifiedDefaultBlock` to check if it's an empty paragraph?

Reply:
Actually, I meant to check the block content only regardless of block type, since there is the case where an empty Heading is followed by a text block, when BACKSPACE, should we remove the empty Heading as well?

Reply:
cc: @ellatrix Do you think we should do the same behavior when empty heading is followed by a text block?

Reply:
Maybe. There's also `isUnmodifiedBlock`, so it would still work.

Reply:
`isUnmodifiedBlock` work well! Update code and re-run e2e testings, thanks for suggestion. Please tell me what you think 😊